sonar-scala-plugin/src/test/java/org/sonarsource/scala/plugin/ScoverageSensorTest.java

```diff
@@ -0,0 +1,173 @@
+    assertThat(context.lineHits(fileKey2, 9)).isNull();
+  }
+
+  @Test
+  public void testLogWhenInvalidAttribute() throws IOException {
+    Path baseDir = COVERAGE_DIR.toAbsolutePath();
+    Path reportPath = baseDir.resolve("badscoverage.xml");
+
+    SensorContextTester context = getSensorContext(reportPath.toString(), "file1.scala", "file2.scala");
+
+    new ScoverageSensor().execute(context);
+
+    String expectedMessage = "Some attributes of statement at line 3 of scoverage report are not present.";
+    assertThat(logTester.logs().contains(expectedMessage)).isTrue();
+  }
+
+  @Test
+  public void testLogWhenUnresolvedInputFiles() throws IOException {
+    Path baseDir = COVERAGE_DIR.toAbsolutePath();
+    Path reportPath = baseDir.resolve("scoverage.xml");
+
+    SensorContextTester context = getSensorContext(reportPath.toString());
+
+    new ScoverageSensor().execute(context);
+
+    String expectedMessage = "Fail to resolve 2 file(s). No coverage data will be imported on the following file(s): /Absolute/Path/To/file1.scala;/Absolute/Path/To/file2.scala";
+
+    assertThat(logTester.logs().contains(expectedMessage)).isTrue();
+  }
+
+  @Test
+  public void testLogInvalidXMLFile() throws IOException {
+    Path baseDir = COVERAGE_DIR.toAbsolutePath();
+    Path reportPath = baseDir.resolve("invalidscoverage.xml");
+
+    SensorContextTester context = getSensorContext(reportPath.toString());
+
+    new ScoverageSensor().execute(context);
+
+    String expectedMessage = String.format("File '" + reportPath.toString() + "' can't be read. ParseError at [row,col]:[2,1]\nMessage: Content is not allowed in prolog.");
+
+    assertThat(logTester.logs().contains(expectedMessage)).isTrue();
+  }
+
+
+  private SensorContextTester getSensorContext(String coverageReportPath, String... fileNames) throws IOException {
+    Path baseDir = COVERAGE_DIR.toAbsolutePath();
+    SensorContextTester context = SensorContextTester.create(baseDir);
+    context.setSettings(new MapSettings());
+    context.settings().setProperty("sonar.scala.coverage.reportPaths", coverageReportPath);
+
+    DefaultFileSystem defaultFileSystem = new DefaultFileSystem(new File(MODULE_KEY));
+    createReportFiles(coverageReportPath, baseDir, defaultFileSystem);
+    for (String fileName : fileNames) {
+      DefaultInputFile inputFile = createInputFile(fileName, fileContent(baseDir, fileName));
+      defaultFileSystem.add(inputFile);
+    }
+
+    context.setFileSystem(defaultFileSystem);
+
+    return context;
+  }
+
+  private void createReportFiles(String coverageReportPath, Path baseDir, DefaultFileSystem defaultFileSystem) throws IOException {
+    String[] coverageReportPaths = coverageReportPath.split(",");
+    for (String reportPath : coverageReportPaths) {
+      reportPath = reportPath.trim();
+      // if report is relative path we create it under fake filesystem
+      if (!Paths.get(coverageReportPath).isAbsolute()) {
```

Comment:
Useless test because `coverageReportPath` are always relative in our unit tests. And I would prefer having a test about _file exists_, to remove the following `try/catch`.